core/modules/replica/DatabaseMySQL.cc

```diff
@@ -0,0 +1,525 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/DatabaseMySQL.h"
+
+// System headers
+#include <sstream>
+#include <stdexcept>
+
+// Third party headers
+#include <boost/lexical_cast.hpp>
+#include <mysql/mysqld_error.h>
+
+// Qserv headers
+#include "lsst/log/Log.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.DatabaseMySQL");
+
+/**
+ * The function is used to comply with the MySQL convention for
+ * the default values of the connection parameters.
+ */
+char const* stringOrNull(std::string const& str) {
+    if (str.empty()) { return 0; }
+    return str.c_str();
+}
```

Comment:
I agree with your statement. Right now tis isn't a problem because the function is used in a well controlled environment. However this may become a problem in the future of someone modified a code calling the function. So, I've chosen to get rid of the function.  There was very little value in the function anyway.